backend/modules/professor/routers/professor_grading.py

```
@@ -0,0 +1,403 @@
+            StudentSimulationInstance.id == instance_id
+        ).first()
+        
+        if not instance:
+            raise HTTPException(status_code=404, detail="Simulation instance not found")
+        
+        # Verify the professor has access to this instance (through cohort assignment)
+        if instance.cohort_assignment:
+            # Check if professor owns the cohort
+            cohort = instance.cohort_assignment.cohort
+            if cohort and cohort.created_by != current_user.id:
+                raise HTTPException(status_code=403, detail="Access denied: You don't have permission to view this submission")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)